src/ray/common/scheduling/resource_set.cc

```diff
@@ -191,7 +191,12 @@ NodeResourceSet::NodeResourceSet(
 
 NodeResourceSet &NodeResourceSet::Set(ResourceID resource_id, FixedPoint value) {
   if (value == ResourceDefaultValue(resource_id)) {
-    resources_.erase(resource_id);
+    // We should still store object_store_memory even if it is 0.
+    if (resource_id == ResourceID::ObjectStoreMemory()) {
```

Comment:
nit: fold this into the top-level condition and flatten the whole thing (no need for nested if-else here)